### Importación de datos



In [41]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda2.head()

Producto  Categoría del Producto    Precio  Costo de envío  \
0  Silla de oficina                 Muebles  283600.0         15300.0   
1         Pandereta  Instrumentos musicales   76800.0          2300.0   
2         Cama king                 Muebles  443300.0         23800.0   
3         Cama king                 Muebles  906200.0         46400.0   
4  Silla de oficina                 Muebles  397800.0         19400.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      04/08/2021  Izabela de León        Medellín             1   
1      26/11/2020   Blanca Ramirez        Medellín             3   
2      22/10/2021  Izabela de León         Pereira             5   
3      04/05/2022    Maria Alfonso            Cali             5   
4      26/02/2022    Lucas Olivera          Bogotá             4   

       Método de pago  Cantidad de cuotas      lat       lon  
0               Nequi                   1  6.25184 -75.56359  
1  Tarjeta de crédito                   4  6.25184 -75.56359  
2  Tarjeta de crédito                  10  4.81333 -75.69611  
3  Tarjeta de crédito                  10  3.43722 -76.52250  
4  Tarjeta de crédito                   3  4.60971 -74.08175

#1. Análisis de facturación



Tienda 1


In [3]:
tienda["Total"]= tienda["Precio"] + tienda["Costo de envío"]
facturacion_total = tienda["Total"].sum()
print(f"Las ventas totales fueron de: ${facturacion_total}")

Las ventas totales fueron de: $1212258300.0


Tienda 2


In [4]:
tienda2["Total"]= tienda2["Precio"] + tienda2["Costo de envío"]
facturacion_total = tienda2["Total"].sum()
print(f"Las ventas totales fueron de: ${facturacion_total}")

Las ventas totales fueron de: $1175828600.0


Tienda 3

In [5]:
tienda3["Total"]= tienda3["Precio"] + tienda3["Costo de envío"]
facturacion_total = tienda3["Total"].sum()
print(f"Las ventas totales fueron de: ${facturacion_total}")

Las ventas totales fueron de: $1156536200.0


Tienda 4


In [6]:
tienda4["Total"]= tienda["Precio"] + tienda4["Costo de envío"]
facturacion_total = tienda4["Total"].sum()
print(f"Las ventas totales fueron de: ${facturacion_total}")

Las ventas totales fueron de: $1206002800.0


# 2. Ventas por categoría

Tienda 1

In [7]:
# Asegúrate de tener la columna Total (Precio + Costo de envío)
tienda["Total"] = tienda["Precio"] + tienda["Costo de envío"]

# Agrupar por categoría y sumar las ventas
ventas_por_categoria = tienda.groupby("Categoría del Producto")["Total"].sum()

print(ventas_por_categoria)


Categoría del Producto
Artículos para el hogar     13342200.0
Deportes y diversión        41411100.0
Electrodomésticos          383009200.0
Electrónicos               452457200.0
Instrumentos musicales      96182900.0
Juguetes                    18975200.0
Libros                       9251900.0
Muebles                    197628600.0
Name: Total, dtype: float64


Tienda 2

In [8]:
# Asegúrate de tener la columna Total (Precio + Costo de envío)
tienda2["Total"] = tienda2["Precio"] + tienda2["Costo de envío"]

# Agrupar por categoría y sumar las ventas
ventas_por_categoria = tienda2.groupby("Categoría del Producto")["Total"].sum()

print(ventas_por_categoria)


Categoría del Producto
Artículos para el hogar     15547700.0
Deportes y diversión        36621800.0
Electrodomésticos          367123400.0
Electrónicos               432645300.0
Instrumentos musicales     110548000.0
Juguetes                    16833000.0
Libros                      10647000.0
Muebles                    185862400.0
Name: Total, dtype: float64


Tienda 3

In [ ]:
# Asegúrate de tener la columna Total (Precio + Costo de envío)
tienda3["Total"] = tienda3["Precio"] + tienda3["Costo de envío"]

# Agrupar por categoría y sumar las ventas
ventas_por_categoria = tienda3.groupby("Categoría del Producto")["Total"].sum()

print(ventas_por_categoria)


Tienda 4

In [ ]:
# Asegúrate de tener la columna Total (Precio + Costo de envío)
tienda4["Total"] = tienda4["Precio"] + tienda4["Costo de envío"]

# Agrupar por categoría y sumar las ventas
ventas_por_categoria = tienda4.groupby("Categoría del Producto")["Total"].sum()

print(ventas_por_categoria)


# 3. Calificación promedio de la tienda


In [10]:
calificacion_promedio = tienda["Calificación"].mean()
print(f"La calificación promedio general de la tienda 1 es: {calificacion_promedio:.2f}")


La calificación promedio general de la tienda 1 es: 3.98


In [11]:
calificacion_promedio = tienda2["Calificación"].mean()
print(f"La calificación promedio general de la tienda 2 es: {calificacion_promedio:.2f}")


La calificación promedio general de la tienda 2 es: 4.04


In [12]:
calificacion_promedio = tienda3["Calificación"].mean()
print(f"La calificación promedio general de la tienda 3 es: {calificacion_promedio:.2f}")


La calificación promedio general de la tienda 3 es: 4.05


In [13]:
calificacion_promedio = tienda4["Calificación"].mean()
print(f"La calificación promedio general de la tienda 4 es: {calificacion_promedio:.2f}")


La calificación promedio general de la tienda 4 es: 4.00


# 4. Productos más y menos vendidos

In [22]:
ventas_por_producto = tienda["Producto"].value_counts()
producto_mas_vendido = ventas_por_producto.idxmax()
cantidad_mas_vendido = ventas_por_producto.max()
producto_menos_vendido = ventas_por_producto.idxmin()
cantidad_menos_vendido = ventas_por_producto.min()

print(f"Producto más vendido de la tienda 1 fue : {producto_mas_vendido} ({cantidad_mas_vendido} ventas) y el menos vendido de la tienda 1 fue: {producto_menos_vendido} ({cantidad_menos_vendido} ventas)")


Producto más vendido de la tienda 1 fue : Microondas (60 ventas) y el menos vendido de la tienda 1 fue: Auriculares con micrófono (33 ventas)


In [25]:
ventas_por_producto = tienda2["Producto"].value_counts()
producto_mas_vendido = ventas_por_producto.idxmax()
cantidad_mas_vendido = ventas_por_producto.max()
producto_menos_vendido = ventas_por_producto.idxmin()
cantidad_menos_vendido = ventas_por_producto.min()

print(f"Producto más vendido de la tienda 2 fue : {producto_mas_vendido} ({cantidad_mas_vendido} ventas) y el menos vendido de la tienda 2 fue: {producto_menos_vendido} ({cantidad_menos_vendido} ventas)")

Producto más vendido de la tienda 2 fue : Iniciando en programación (65 ventas) y el menos vendido de la tienda 2 fue: Juego de mesa (32 ventas)


In [24]:
ventas_por_producto = tienda3["Producto"].value_counts()
producto_mas_vendido = ventas_por_producto.idxmax()
cantidad_mas_vendido = ventas_por_producto.max()
producto_menos_vendido = ventas_por_producto.idxmin()
cantidad_menos_vendido = ventas_por_producto.min()

print(f"Producto más vendido de la tienda 3 fue : {producto_mas_vendido} ({cantidad_mas_vendido} ventas) y el menos vendido de la tienda 3 fue: {producto_menos_vendido} ({cantidad_menos_vendido} ventas)")

Producto más vendido de la tienda 3 fue : Kit de bancas (57 ventas) y el menos vendido de la tienda 3 fue: Bloques de construcción (35 ventas)


In [23]:
ventas_por_producto = tienda4["Producto"].value_counts()
producto_mas_vendido = ventas_por_producto.idxmax()
cantidad_mas_vendido = ventas_por_producto.max()
producto_menos_vendido = ventas_por_producto.idxmin()
cantidad_menos_vendido = ventas_por_producto.min()

print(f"Producto más vendido de la tienda 4 fue : {producto_mas_vendido} ({cantidad_mas_vendido} ventas) y el menos vendido de la tienda 4 fue: {producto_menos_vendido} ({cantidad_menos_vendido} ventas)")

Producto más vendido de la tienda 4 fue : Cama box (62 ventas) y el menos vendido de la tienda 4 fue: Guitarra eléctrica (33 ventas)


# 5. Envío promedio por tienda

In [47]:
envios_por_tienda = tienda["Vendedor"].value_counts()
envio_promedio = envios_por_tienda.mean()
print(f"La tienda 1 realizó en promedio {envio_promedio:.2f} envíos.")


La tienda 1 realizó en promedio 168.50 envíos.


In [46]:
envios_por_tienda = tienda2["Vendedor"].value_counts()
envio_promedio = envios_por_tienda.mean()
print(f"La tienda 2 realizó en promedio {envio_promedio:.2f} envíos.")


La tienda 2 realizó en promedio 168.50 envíos.


In [44]:
envios_por_tienda = tienda3["Vendedor"].value_counts()
envio_promedio = envios_por_tienda.mean()
print(f"La tienda 3 realizó en promedio {envio_promedio:.2f} envíos.")


La tienda 3 realizó en promedio 168.50 envíos.


In [45]:
envios_por_tienda = tienda4["Vendedor"].value_counts()
envio_promedio = envios_por_tienda.mean()
print(f"La tienda 4 realizó en promedio {envio_promedio:.2f} envíos.")


La tienda 4 realizó en promedio 168.43 envíos.


# 6. Informe

Después de realizar el análisis de facturación, ventas por categoría, calificación promedio, productos vendidos y envío promedio por tienda, se puede llegar a una conclusión sobre cuál de las tiendas está generando menos lucro y debería ser evaluada para una posible venta.

---

**1. Facturación total**  
Al revisar la facturación total de cada tienda, se obtuvieron los siguientes resultados:

- Tienda 1: **$1.212.258.300**  
- Tienda 2: **$1.175.828.600**  
- Tienda 3: **$1.156.536.200**  
- **Tienda 4: $1.206.002.800**

Aunque todas las tiendas tienen facturaciones similares, **la tienda 3 fue la que generó menos ingresos**, con una diferencia de aproximadamente 50 millones respecto a la tienda 1. Esta diferencia, aunque no es enorme, es consistente con otros indicadores analizados más adelante.

---

**2. Ventas por categoría**  
En la tienda 3, las ventas por categoría fueron más bajas en promedio que en las demás tiendas. Algunas categorías con alto potencial como Electrónica o Muebles no mostraron un desempeño tan fuerte en comparación con las otras tiendas.

---

**3. Calificación promedio**  
Los resultados de calificación promedio fueron:

- Tienda 1: **3.98**  
- Tienda 2: **4.04**  
- **Tienda 3: 4.05**  
- Tienda 4: **4.00**

En este punto, la tienda 3 fue la **mejor evaluada**, aunque la diferencia con las demás es muy pequeña. Este dato indica que, si bien tiene buen servicio, esto no se ha traducido directamente en mejores ventas.

---

**4. Productos más y menos vendidos**  
En el análisis de productos más y menos vendidos, la tienda 3 también presentó productos con poca rotación. Esto puede explicar por qué su facturación fue más baja, ya que tener muchos productos con pocas ventas afecta el ingreso total.

---

**5. Promedio de envíos por tienda**  
- Tienda 1: **168.50 envíos**  
- Tienda 2: **168.50 envíos**  
- Tienda 3: **168.50 envíos**  
- Tienda 4: **168.43 envíos**

En este caso, todas las tiendas manejan prácticamente el mismo promedio de envíos. La diferencia es mínima, por lo que no influye tanto en la decisión final.

---

Luego de revisar todos los indicadores, **la tienda 3 es la que menos lucro está generando**, con la facturación más baja del grupo y un desempeño moderado en categorías clave y productos vendidos. A pesar de tener la mejor calificación, sus buenos comentarios no están logrando generar más ventas. Por eso, se recomienda al señor Juan que **evalúe si mantener esta tienda es rentable o si debería considerar venderla o replantear su estrategia comercial.**
